spec/controllers/admin/videos_controller_spec.rb

```diff
@@ -1,64 +1,81 @@
 require 'spec_helper'
 
 describe Admin::VideosController do
+
   describe "GET new" do
+
     it_behaves_like 'require_sign_in' do
       let(:action) { get :new }
     end
+
     it_behaves_like 'require_admin' do
       let(:action) { get :new }
     end
+
     it "sets @video to a new video record with admin" do
       set_current_admin
       get :new
-      expect(assigns(:video)).to be_new_record
-      expect(assigns(:video)).to be_instance_of(Video)
+      expect(assigns(:video)).to be_a_new(Video)
     end
+
   end
 
   describe "POST create" do
+
     it_behaves_like 'require_sign_in' do
       let(:action) { post :create }
     end
+
     it_behaves_like 'require_admin' do
       let(:action) { post :create }
     end
 
     context "with valid input" do
+
       before do
         set_current_admin
         category = Fabricate(:category)
-        post :create, video: { title: "Paul", description: "A funny alien movie.", categories: category.id}
+        post :create, video: { title: "Paul", description: "A funny alien movie.", category_id: category.id}
       end
+
       it "redirects to the add new video page" do
         expect(response).to redirect_to(new_admin_video_path)
       end
+
       it "creates a new video" do
         expect(Category.last.videos.count).to eq(1)
       end
+
       it "sets the flash[:success]" do
         expect(flash[:success]).to be_present
       end
+
     end
 
     context "with invalid input" do
+
       before do
         set_current_admin
         category = Fabricate(:category)
-        post :create, video: { description: "A funny alien movie.", categories: category.id}
+        post :create, video: { description: "A funny alien movie.", category_id: category.id}
```

Comment:
I change to category: category and there's one test failed.

Since I user videos_params to add new video record in the controller, so will need to pass the category_id for the videos_params to correctly process the request.

If I want to use the object to object assignment in the test, then I will have to use the object to object in my controller, right?

```app/videos_controller.rb
def create
  @video = Video.new(video_params)
  ...
end

private

def video_params
  params.require(:video).permit(:title, :category_id, :description, :large_cover, :small_cover, :video_url)
end
```